### ISSUE TITLE
Is it possible to recreate antsMotionCorr's 5D displacement field (-w) from multiple antsRegistration 4D displacement fields?

### ISSUE THREAD

**Issue by bfranx** [ANTsX/ANTs#1301]
I'm following [this tutorial](https://stnava.github.io/fMRIANTs/) to register a challenging functional imaging time series to a common fixed space. There is some slow but noticeable motion in my data: normally one would use antsMotionCorr to correct for this, but it consistently makes things worse in my dataset. 
Disassembling the time series and doing linear motion correction piece-by-piece with antsRegistration in a loop does work, however I'm unable to apply all the transforms to my original uncorrected dataset in one go and continue the fMRI pipeline.

My approach was to recreate the displacement field that antsMotionCorr supplies (-w), by assembling all the affine transforms (converted to non-linear displacement fields as discussed [here](https://sourceforge.net/p/advants/discussion/840261/thread/da10cb34/)) using ImageMath TimeSeriesAssemble. So far so good for the fourth dimension, but this doesn't recreate the fifth component. Unsurprisingly, my 'contrived' displacement field is about 6x smaller in size compared to the 5D displacement field output by antsMotionCorr, and antsApplyTransform terminates without results.

How should I go about this?

**Comment by cookpa** [comment]
Can you please specify exactly what commands you are running? Better yet, with example data. But at least the commands.

**Comment by bfranx** [comment]
Sure thing, [I prepared a toy example and a shell script](https://surfdrive.surf.nl/files/index.php/s/JxlzdAkpT9zVAcu)

The output data is already there: `*aRmc_assembled.nii.gz` are the concatenated images from antsRegistration, compare that to `*aMC.nii.gz`, the output from antsMotionCorr to see the difference in performance (even though the -t arguments are mostly the same). `*aRmc_Warped.nii.gz` was my attempt to motion correct the original data with motion with the 'rebuilt' displacement field.

**Comment by cookpa** [comment]
Thanks I took a look at the script. The `TimeSeriesAssemble` command is meant to stitch scalar data into a time series, I don't think that handles vector images. 

Is there a reason you can't warp each volume of `$data` and then call `ImageMath 4 warped4D.nii.gz TimeSeriesAssemble 0 $tr ...`?

I tried to see why `antsMotionCorr` gives you such different results. The only difference I see is you allow a maximum of 100 iterations at full resolution in `antsMotionCorr` and only 75 in `antsRegistration`. Also, the convergence threshold is hard coded in `antsMotionCorr` to 10E-7 and you use 10E-6 in `antsRegistration`. So if it's doing way less iterations in `antsRegistration`, it might be overfitting that is correctable in `antsMotionCorr` by reducing the max iterations.

**Comment by bfranx** [comment]
Thanks for your assistance. 

I suspected the same thing about antsMotionCorr since it doesn't allow you to tweak convergence thresholds. I suppose it wasn't developed with this kind of data in mind.

I rewrote the pipeline as you suggested, I'm aligning each volume to the fixed spaced individually and stitch them together over there. It seems to work so far, but it leaves me wondering what the additional 5th component of antsMotionCorr does and where it comes from. To me it implies antsMotionCorr is more than a wrapper for antsRegistration on time series.
